text/0019-stricter-validation.md

```diff
@@ -0,0 +1,77 @@
+# Standalone strict validation
+
```

Comment:
After hashing out these changes in the Rust I'm convinced that these changes make sense.

There was one cases I encountered where the new definition was stricter in a way I didn't anticipate. If you have an expression like `<strict_error> || true`, the old strict typechecker would see that `|| true` makes the whole expression true, so it got rewritten to `true` without strict typechecking, but now strict typechecking is applied to the LHS before we know RHS is `true`, so strict type errors are reported.

This is another instance where the new behavior is not bad. In this case it's because we used to have permissive type errors reported regardless of the `|| true` (it would be unsound to do otherwise), but strict type error reporting did depend on the right operand. The new behavior is more consistent.